### DISCUSSION TITLE
How to disable t/T and f/F overrides?

### DISCUSSION THREAD

**Discussion by augustocdias** [ggandor/lightspeed.nvim#110] [Q&A]
I'm generally fine with the idea of what they do but they make the `.` stop working or making it with really confusing behavior.

I use a lot `cf<CHAR>`, replace the text, next occurrence and `.` and lightspeed just breaks that flow. I would like to be able to disable those overrides and have only the jump with s/S which is what this plugin is most useful for me.

**Comment by ggandor** [comment]
> they make the . stop working or making it with really confusing behavior.

Could you elaborate? If they do not work with `.` like the native `f/t`, then please raise an issue instead, because it is a bug then. But I'm not experiencing any such problems.

**Reply by augustocdias** [reply]
When I press `.` it doesn't repeat the operation. It waits for the `CHAR` again for `f/t`. So instead of using `cf<CHAR>` and then `.` I have to do `.<CHAR>`.

I had some cases that pressing `.` caused it to go lines ahead instead of the match in the same line
I had a text like

```
fn test_my_<CURSOR>function_error()
```
I had just executed `ct_` in another line and pressing `.` caused it to go untile lines down instead of the match that should be near the `error` part.

I would prefer to disable that part of the plugin since it is not useful for me...

**Reply by ggandor** [reply]
> When I press . it doesn't repeat the operation. It waits for the CHAR again for f/t. So instead of using cf<CHAR> and then . I have to do .<CHAR>.

This is not how dot-repeat is intended to work. I suspect you don't have [vim-repeat](https://github.com/tpope/vim-repeat) installed (https://github.com/ggandor/lightspeed.nvim#-getting-started), plugins like Lightspeed usually need that as a dependency for the `.` functionality. With that, `.`-ing f/t/z/x should work as you would expect. But if you're getting this behaviour with repeat-vim enabled, then definitely raise an issue, we should fix some bugs in that case.

(That said, you can disable f/t: https://github.com/ggandor/lightspeed.nvim#disabling-features, but please don't do that before talking this through.)

**Reply by augustocdias** [reply]
I don't have vim-repeat installed and I don't want to. I want to disable those features because they're not useful to me. I never use f/t across lines, so that feature is not important for me.

**Reply by augustocdias** [reply]
What are all the keys that lightspeed overrides?

**Reply by ggandor** [reply]
> I don't have vim-repeat installed and I don't want to. I want to disable those features because they're not useful to me. I never use f/t across lines, so that feature is not important for me.

Oh, okay, then just `unmap f | unmap F | unmap t | unmap T` (there's no other way at the moment). The opening sentence "I'm generally fine with the idea of what they do >>but<< they make the . stop working..." gave me the impression that ideally you would want to preserve the multi-line f/t feature too.

**Reply by ggandor** [reply]
> What are all the keys that lightspeed overrides?

https://github.com/ggandor/lightspeed.nvim#keymaps

**Reply by augustocdias** [reply]
I'm fine with it as long as it works out of the box 🤣 (without deps). It is still very unlikely I'll use it though. So unmaping is a better solution for me.

I love the "hoping" feature with S/s and thats what I use very often with it. Perhaps I should dig into the docs to learn more about what it can do.

**Reply by ggandor** [reply]
> I'm fine with it as long as it works out of the box 🤣 (without deps)

Come on, that's the whole point of developing the ecosystem together - the very point of auxiliary plugins like vim-repeat is that other authors should not reinvent the wheel :)

**Reply by augustocdias** [reply]
I agree with you but after I moved my setup to lua and only lua based plugins I've became biased and started to avoid any plugin written in vimscript. And since I'd have it only for using features from lightspeed that are not important to me, I chose not to.

One question. How do I add those unmaps to my setup? I tried adding them in my lua init files and I got errors that there's no mapping for the keys. I suppose lightspeed is sourced after it. So there's no way to automate this?

**Reply by ggandor** [reply]
Just move those `unmap` lines to somewhere after the packer setup line then.

**Reply by augustocdias** [reply]
First I tried adding in the config function property from lightspeed packer entry, but it didn't work. Then I tried adding at the end of everything in my setup and it also didn't work... If I execute the command after I open it, it works fine

**Reply by augustocdias** [reply]
Wouldn't it be better to have an option in the setup to disable this? I'd be willing to help with a PR

**Reply by ggandor** [reply]
I don't use packer, but couldn't it be that packer is somehow set to automagically lazy-load lightspeed, on some event?

If all else fails, you could just map something (unused) to the `<plug>` keys instead, _before_ sourcing lightspeed (that will prevent lightspeed from creating the corresponding mappings):

```vim
map <a-s-f9> <Plug>Lightspeed_f
map <a-s-f10> <Plug>Lightspeed_F
" etc.
```

**Reply by ggandor** [reply]
> Wouldn't it be better to have an option in the setup to disable this? I'd be willing to help with a PR

https://github.com/ggandor/lightspeed.nvim/issues/11

It would imply a breaking change, because calling `setup()` would then be mandatory for everyone who want the default keymaps. And I consider such requests - i.e., not wanting to use `f/t` or `s/x` at all - exceptional, so I'm not too keen on providing this feature as a built-in option.

I understand that a few users will always have exceptional needs, and that's totally OK, but as I wrote in the above linked thread, 
> The assumption is that the default mappings are sane and carefully thought out, and 95% of the users want to use the provided features as they are. The others could do some extra mileage I guess.

**Reply by augustocdias** [reply]
Why would it be mandatory if you make all enabled by default?

**Reply by ggandor** [reply]
Just think about it: if we map the keys by default, i.e. without needing to call `setup()`, it means that we do that unconditionally when sourcing the plugin, so setting a flag via `setup` could make no difference. All we could do is to unmap the already mapped keys in that case.
